docs/sidebars.js

```diff
@@ -1,152 +1,80 @@
-/**
- * Creating a sidebar enables you to:
- - create an ordered group of docs
- - render a sidebar for each doc of that group
- - provide next/previous navigation
-
- The sidebars can be generated from the filesystem, or explicitly defined here.
-
- Create as many sidebars as you want.
- */
-
 // @ts-check
+import fs from 'node:fs';
 
-/** @type {import('@docusaurus/plugin-content-docs').SidebarsConfig} */
-const sidebars = {
-  // By default, Docusaurus generates a sidebar from the docs folder structure
-  // tutorialSidebar: [{type: 'autogenerated', dirName: '.'}],
-
-  // But you can create a sidebar manually
-  sidebarNode: [
+async function sleep(timeout) {
+  return new Promise((res) => {
+    setTimeout(res, timeout);
+  });
+}
+async function waitAndImportFile(file) {
+  while (!fs.existsSync(file)) {
+    await sleep(50);
+  }
+  return require(file);
+}
+/**
+ * Generate a sidebar object for use as reference documentation sidebar entry for each package.
+ * @param {string} pkg the sub-package to generate refdoc options for
+ */
+async function packageSidebarEntry(pkg) {
+  return {
+    type: 'category',
+    label: `@slack/${pkg}`,
+    link: {
+      type: 'doc',
+      id: `reference/${pkg}/index`,
+    },
+    items: await waitAndImportFile(`./content/reference/${pkg}/typedoc-sidebar.cjs`),
+  };
+}
+/**
+ * Generate list of sidebar objects for docs site.
+ * @param {string[]} packages list of sub-package to generate refdoc sidebar entries for
+ */
+export default async function sidebarGenerator(packages) {
+  const apiReferenceItems = [];
+  for (const pkg of packages) {
+    apiReferenceItems.push(await packageSidebarEntry(pkg));
+  }
+  return [
     {
       type: 'doc',
       id: 'index', // document ID
       label: 'Node Slack SDK', // sidebar label
       className: 'sidebar-title',
     },
-    'getting-started',
-    'typescript',
-    'packages/web-api',
-    'packages/oauth',
-    'packages/rtm-api',
-    'packages/webhook',
-    'packages/socket-mode',
+    { type: 'doc', id: 'getting-started' },
```

Comment:
Moving to the more fine-control approach of [providing our own asynchronous sidebar generator function](https://typedoc-plugin-markdown.org/plugins/docusaurus/sidebar#injecting-into-autogenerated-sidebar), for some reason, requires the output of the function to be the more verbose list-of-objects for sidebar items - I'd get errors otherwise. So, had to move away from the the convenience string items 🤷